libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/push/OnNotifiableEventReceived.kt

```diff
@@ -32,10 +35,23 @@ interface OnNotifiableEventReceived {
 class DefaultOnNotifiableEventReceived @Inject constructor(
     private val defaultNotificationDrawerManager: DefaultNotificationDrawerManager,
     private val coroutineScope: CoroutineScope,
+    private val matrixClientProvider: MatrixClientProvider,
+    private val featureFlagService: FeatureFlagService,
 ) : OnNotifiableEventReceived {
     override fun onNotifiableEventReceived(notifiableEvent: NotifiableEvent) {
         coroutineScope.launch {
+            subscribeToRoomIfNeeded(notifiableEvent)
             defaultNotificationDrawerManager.onNotifiableEventReceived(notifiableEvent)
         }
     }
+
+    private fun CoroutineScope.subscribeToRoomIfNeeded(notifiableEvent: NotifiableEvent) = launch {
+        if (!featureFlagService.isFeatureEnabled(FeatureFlags.SyncOnPush)) {
+            return@launch
+        }
+        val client = matrixClientProvider.getOrRestore(notifiableEvent.sessionId).getOrNull() ?: return@launch
+        client.getRoom(notifiableEvent.roomId)?.use { room ->
+            room.subscribeToSync()
```

Comment:
I think that might be instantiating a timeline too?

Reply:
Yes the goal is also to create and cache a timeline before clicking the notification